azurerm/validators.go

```diff
@@ -74,3 +75,114 @@ func validateIso8601Duration() schema.SchemaValidateFunc {
 		return
 	}
 }
+
+func validateAzureTimeZone() schema.SchemaValidateFunc {
```

Comment:
Can we rename this to `validateAzureVmTimeZone()`? other APIs might accept different values (the automation one accepts UTC+10 for instance)

Reply:
I'm trying to make it a general validation rule which would also be used in other timezones (like [the one here](https://github.com/terraform-providers/terraform-provider-azurerm/blob/94d287bcb11c3f6cb1ce238d95f9d6dc67168577/azurerm/resource_arm_automation_schedule.go#L72))

Reply:
given the Azure API's vary so much there's no need to make this generic; in Go the general convention is to value duplication over a completely DRY codebase with extra complexities for switch statements

Reply:
The property you linked to accepts UTC+10 in it API as well as the portal sets it to an entirely different timezone format. For example PST is "America/Vancouver"

Reply:
Changed.